casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -215,29 +214,51 @@ object ProtoUtil {
 
   def unsignedBlockProto(body: Body,
                          header: Header,
-                         justifications: Seq[Justification]): BlockMessage =
+                         justifications: Seq[Justification]): BlockMessage = {
+    val hash = hashUnsignedBlock(header, justifications)
+
     BlockMessage()
-      .withBlockHash(protoHash(header))
+      .withBlockHash(hash)
       .withHeader(header)
       .withBody(body)
       .withJustifications(justifications)
+  }
+
+  def hashUnsignedBlock(header: Header, justifications: Seq[Justification]) = {
+    val items = header.toByteArray +: justifications.map(_.toByteArray)
+    hashByteArrays(items: _*)
+  }
+
+  def hashSignedBlock(header: Header, sender: ByteString, sigAlgorithm: String, seqNum: Int) =
+    hashByteArrays(header.toByteArray, sender.toByteArray, sigAlgorithm.getBytes, Array(seqNum.toByte))
```

Comment:
Is `Array(seqNum.toByte)` Scala specific? Mabye use protobuf's toByteArray instead? We want to allow for non-Scala implementations to be able to reproduce this block hashing process.